lib/SimpleSAML/Logger.php

```diff
@@ -393,6 +398,9 @@ private static function defer($level, $message, $stats)
             register_shutdown_function([self::class, 'shutdown']);
             self::$shutdownRegistered = true;
         }
+
+        // Set initializing to true > all went well!
+        self::$initializing = true;
```

Comment:
There is something weird here, and it's difficult to follow the code with the current name of the property and the way it's assigned values. I think `$initializing` should default to `false`, and set to `true` when we call `createLoggingHandler()`, and back to `false` after we're done there.